### PR TITLE
Moved exclude check to step 2

### PR DESCRIPTION
**Pull request by ConnorSkees**
Not to spam you with pull requests, but is this what you were thinking?

### CODE REVIEW

**Comment by s0md3v** [comment]
Great!
Can you confirm that it's working as intended?

**Comment by ConnorSkees** [comment]
Yes everything works exactly as intended. --exclude .* and --exclude http.* return 0 urls while --exclude 
\d and --exclude "" return the same as without the exclude flag.

**Comment by s0md3v** [comment]
It's not working on my end.

**Comment by ConnorSkees** [comment]
With no flag:
image


Flag of .*
image


Flag of "thisdoesnotexist"
image

Flag of .*teach.* (note that only one link is removed)
image

**Comment by s0md3v** [comment]
Doesn't work :')

screenshot_2018-08-03_02-46-57

**Comment by ConnorSkees** [comment]
mm do you want to exclude keywords or regex's? with the regex, change to .\*questions

File: photon.py [hunk 1]
```diff
@@ -16,7 +16,6 @@
     from urllib.parse import urlparse # for python3
     python2, python3 = False, True
 except ImportError:
-    input = raw_input
```
Comment (s0md3v) [inline photon.py]: Why did you remove this?

File: photon.py [hunk 2]
```diff
@@ -2,14 +2,14 @@
 # -*- coding: utf-8 -*-
 
 # Let's import what we need
+import argparse
 import os
 import sys
-import time
 import random
-import warnings
-import argparse
```
Comment (s0md3v) [inline photon.py]: Why did you re-arrange imports? Any specific reason?
Because I kept them in order according to their length so it looks less cluttered.
I have followed the same schema wherever possible.

**Comment by ConnorSkees** [comment]
Sorry, my bad I added it back in with Update photon.py (4c60e1b)

File: photon.py (hunk 2 above)
Comment (ConnorSkees) [reply photon.py]: I did them by standard/3rd party in alphabetical order to quickly search for specific modules. If you disagree and think it makes it worse I totally understand :) its just how I prefer imports

**Comment by s0md3v** [comment]
It still doesn't work :)

screenshot_2018-08-03_12-55-09

**Comment by ConnorSkees** [comment]
I think the issue is that although it isn't crawling the links, it is still adding them to the links file. Is it ok to just use `remove_regex()` on the list before they are exported?

**Comment by ConnorSkees** [comment]
Now testing this against the links file, links containing the regex are not added.
